### ISSUE TITLE
some question about cyclegan.py

### ISSUE THREAD

**Issue by blight19** [eriklindernoren/PyTorch-GAN#63]
I'm a little confused about why use mse_loss in GAN loss and use L1 loss in Cycle loss and Identity loss,and i didn't find this in the paper.
And the second question is why use the fake_A_buffer.push_and_pop()  ,it seems to do something like if the len<50 do nothing,and when len>50,the part of >50 do the random choice the sample? i really confused about this

**Comment by morawi** [comment]
Well, I think you can try whatever loss function, this is a no-big-deal.
As for the buffer, it will start random pooling after it is filled (max-size=50). This won't affect the performance as one usually has tens of thousands of iterations, or even more.

**Comment by SITUSITU** [comment]
> 

Can you be more specific, please? My personal understanding about what the buffer do is that the first 50 fake sample from the generator are fed into the discriminator as usual, but in the meantime they are stored in a list.
When the generator generates the 51st fake sample, the newest fake sample has a 50% chance to replace a element in the list randomly. then the replaced element in the list is fed into the discriminator. There is also a 50% chance that the newest fake sample is fed into the discriminator directly, which mean the list is not change.
 if my understanding is correct? What is the purpose of this.
